Modules/BookingManager/classes/class.ilObjBookingPoolGUI.php

```diff
@@ -1689,23 +1689,33 @@ public function rsvConfirmDeleteObject()
     public function rsvDeleteObject()
     {
         global $DIC;
-        if ($DIC->http()->request()->getParsedBody()['rsv_ids']) {
+        $get = $DIC->http()->request()->getParsedBody()['rsv_ids'];
+        if ($get) {
             include_once 'Modules/BookingManager/classes/class.ilBookingReservation.php';
-            foreach (explode(',', $DIC->http()->request()->getParsedBody()['rsv_ids']) as $id) {
-                $obj = new ilBookingReservation($id);
-                if (!$this->checkPermissionBool("write")) {
+            foreach (explode(',', $get) as $id) {
+                $res = new ilBookingReservation($id);
+                $obj = new ilBookingObject($res->getObjectId());
+                $ref_ids = ilObject::_getAllReferences($obj->getPoolId());
+                $access = false;
+                foreach ($ref_ids as $ref_id) {
+                    if ($this->access->checkAccess("write", "", $ref_id)) {
+                        $access = true;
+                        break;
+                    }
+                }
```

Comment:
You only need to check
- write permission for the current ref id and
- does the $obj->getPoolId() match the current pool id.

No need to get all the references.